modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DefaultMutableConfigurationStats.java

```diff
@@ -100,4 +109,22 @@ public int discountRunningSource() {
     updateLastUsed();
     return runningSources.decrementAndGet();
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public int addActiveComponent() {
+    updateLastUsed();
```

Comment:
if all the other methods are deprecated in favor of these new set of methods, then the other ones should delegate into this one instead of having what's essentially repeated code.

Reply:
Wouldn't that break the contract for getRunningSources() and getInflightOperations() (see https://github.com/mulesoft/mule-extensions-api/pull/750/files) ? If addRunningSource() delegates to addActiveComponent() then the activeComponents counter would be raised and the runningSources would remain at 0. Then asking for getRunningSources() would return 0.
The deprecated components should preserve their behavior and thus their counters, I think.
I do agree with removing the checks mentioned in other comments.

Reply:
yes, you're right